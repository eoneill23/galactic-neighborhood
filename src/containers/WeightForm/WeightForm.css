.calculation-label {
  text-decoration: underline;
}