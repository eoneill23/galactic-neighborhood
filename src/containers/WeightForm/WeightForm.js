import React, { Component } from 'react';
import { connect } from 'react-redux';
import './WeightForm.css';
import PropTypes from 'prop-types';

export class WeightForm extends Component {
  constructor() {
    super();
    this.state = {
      number: 0
    }
  }

  handleChange = e => {
    this.setState({[e.target.name]: e.target.value })
  }

  handleSubmit = e => {
    e.preventDefault();
    this.props.calculateWeight(this.state.number, this.props.multiplier);
    this.clearInputs();
  }

  clearInputs = () => {
    this.setState({ number: 0 })
  }

  render() {
    return (
      <form>
        <p className='calculation-label'>What would your weight on this planet be? Enter your weight in the field below and click 'submit' to find out!</p>
        <input
          type='number'
          placeholder='Enter your weight (lbs)'
          name='number'
          value={this.state.number}
          onChange={this.handleChange}
        />
        <button
        onClick={(e) => this.handleSubmit(e)}
        >
          Submit
        </button>
        {this.props.calculation > 0 && <p>Your weight on this planet would be: {this.props.calculation.toFixed(2)} pounds!</p>}
      </form>
    )
  }
}

export const mapStateToProps = state => ({
  calculation: state.calculation
});

export default connect(mapStateToProps)(WeightForm);

WeightForm.propTypes = {
  calculation: PropTypes.number,
  calculateWeight: PropTypes.func,
  multiplier: PropTypes.number
}