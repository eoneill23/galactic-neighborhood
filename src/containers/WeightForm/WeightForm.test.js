import React from 'react';
import { shallow } from 'enzyme';
import { WeightForm, mapStateToProps } from './WeightForm';

describe('WeightForm', () => {
  let wrapper, mockCalculateWeight;

  beforeEach(() => {
    mockCalculateWeight = jest.fn();
    wrapper = shallow(
      <WeightForm 
      calculation={100}
      multiplier={2}
      calculateWeight={mockCalculateWeight}
      />
    );
  });

  it('should match the snapshot with all data passed in correctly', () => {

    expect(wrapper).toMatchSnapshot();
  });

  
});