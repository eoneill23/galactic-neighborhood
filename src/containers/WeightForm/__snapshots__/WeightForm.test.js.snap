// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeightForm should match the snapshot with all data passed in correctly 1`] = `
<form>
  <p>
    What would your weight on this planet be? Enter your weight in the field below and click 'submit' to find out!
  </p>
  <input
    name="number"
    onChange={[Function]}
    placeholder="Enter your weight (lbs)"
    type="number"
    value={0}
  />
  <button
    onClick={[Function]}
  >
    Submit
  </button>
  <p>
    Your weight on this planet would be: 
    100.00
     pounds!
  </p>
</form>
`;
